app/internal/events/calculate_orders_repository.py

```diff
@@ -0,0 +1,42 @@
+from abc import ABC
+from datetime import datetime
+from typing import List, Optional
+
+from app.domain.repositories.calculate_orders_repository_interface import CalculateOrdersRepositoryInterface
+from app.models import ProductIngredientAssociation, OrderItem
+
+
+class CalculateOrdersRepository(CalculateOrdersRepositoryInterface):
+
+    async def get_items_by_product(self, product_id: str) -> Optional[List[ABC]]:
+        """
+        Method returns items for recipe
+
+        Args:
+            product_id (str): id of product for recipe
+
+        Returns:
+            Optional[List[ABC]]: List of recipes by product_id or None
+        """
+        return await ProductIngredientAssociation.get_items_by_product(product_id)
```

Comment:
as i think its better to return consistent data. So if there is no items - just return empty array